--- pastix.h.orig	2024-04-23 01:05:29.283386000 +0200
+++ pastix.h	2024-04-23 01:08:06.911723000 +0200
@@ -46,6 +46,11 @@
 		ITG *neq, ITG *nzs, ITG *symmetryflag, ITG *inputformat,
 		ITG *jq, ITG *nzs3);
 
+void pastix_factor_main_cp(double *ad, double *au, double *adb, double *aub, 
+        double *sigma,ITG *icol, ITG *irow, 
+                ITG *neq, ITG *nzs, ITG *symmetryflag, ITG *inputformat,
+                ITG *jq, ITG *nzs3);
+
 void pastix_factor_main_generic(double *ad, double *au, double *adb, double *aub, 
         double *sigma,ITG *icol, ITG *irow, 
 		ITG *neq, ITG *nzs, ITG *symmetryflag, ITG *inputformat,
@@ -60,6 +65,8 @@
 ITG pastix_solve(double *b,ITG *neq,ITG *symmetryflag,ITG *nrhs);
 
 ITG pastix_solve_as(double *x, ITG *neq,ITG *symmetryflag,ITG *nrhs);
+
+ITG pastix_solve_cp(double *x, ITG *neq,ITG *symmetryflag,ITG *nrhs);
 
 ITG pastix_solve_generic(double *x, ITG *neq,ITG *symmetryflag,ITG *nrhs);
 
