--- pastix.c.orig	2024-08-05 19:13:20.783702850 +0200
+++ pastix.c	2024-10-12 18:56:54.777791000 +0200
@@ -938,7 +938,6 @@
 void pastix_cleanup(ITG *neq,ITG *symmetryflag){
 	if( redo && !firstIter ){
         if(spm->values == aupastix) spm->values = NULL;
-        if(spm->values == spm->valuesGPU) spm->valuesGPU = NULL;
         if(spm->colptr == icolpastix) spm->colptr = NULL;
         if(spm->rowptr == irowpastix) spm->rowptr = NULL;
 		spmExit( spm );
