--- bcsc/bcsc_zinit.c.orig	2024-03-23 21:09:17.394293000 +0100
+++ bcsc/bcsc_zinit.c	2024-03-24 00:09:31.884070000 +0100
@@ -26,8 +26,8 @@
 						pastix_int_t* 		colptrIn, 
 						pastix_int_t* 		rowptrIn,
 						pastix_int_t**		rowPrediction,
-						cuDoubleComplex*	valuesIn, 
-						cuDoubleComplex* 	valuesOut){
+						pastix_complex64_t*	valuesIn, 
+						pastix_complex64_t* 	valuesOut){
 	
 	if(*rowPrediction == NULL){
 		pastix_int_t* temp = (pastix_int_t*) calloc(n, sizeof(pastix_int_t));
@@ -52,12 +52,12 @@
 void permute_z_Matrix(pastix_int_t		n,
 					  pastix_int_t* 	colptrIn,
 					  pastix_int_t* 	rowptrIn, 
-					  cuDoubleComplex* 	valuesIn, 
+					  pastix_complex64_t* 	valuesIn, 
 					  pastix_int_t* 	perm, 
 					  pastix_int_t* 	peri, 
 					  pastix_int_t* 	colptrOut, 
 					  pastix_int_t* 	rowptrOut, 
-					  cuDoubleComplex*  valuesOut,
+					  pastix_complex64_t*  valuesOut,
 					  pastix_int_t* 	sorttab){
 	
 	colptrOut[0] = 1;
