--- bcsc/bcsc_z.h.orig	2024-04-22 19:47:27.890452000 +0200
+++ bcsc/bcsc_z.h	2024-04-22 19:47:49.766305000 +0200
@@ -18,7 +18,6 @@
 
 
 
-#if defined(PASTIX_WITH_CUDA)
 void transpose_z_Matrix(pastix_int_t 		n,
 						pastix_int_t* 		colptrIn, 
 						pastix_int_t* 		rowptrIn, 
@@ -36,7 +35,6 @@
 					  pastix_int_t* 	rowptrOut, 
 					  cuDoubleComplex*  valuesOut,
 					  pastix_int_t* 	sorttab);
-#endif
 					  
 /**
  * @addtogroup bcsc_internal
