--- Conversion.py.orig	2024-04-21 20:35:34.492459000 +0200
+++ Conversion.py	2024-04-21 20:36:39.035141000 +0200
@@ -245,6 +245,7 @@
         if sub_type != 'tracing' :
           replace = replace.replace('\(','(');
           replace = replace.replace('\)',')');
+        replace = replace.replace('\s*',' ');
         data = re.sub(search, replace, data);
       except:
         print('Bad replacement pair ',i,'in',sub_type);
