--- sopalin/pastix_task_sopalin.c.orig	2024-03-23 20:54:13.699823000 +0100
+++ sopalin/pastix_task_sopalin.c	2024-03-23 20:55:30.936111000 +0100
@@ -471,8 +471,10 @@
         sopalin_data.solvmtx = pastix_data->solvmatr;/*
         sopalin_data.gpu_block_dgemm = 0;
         sopalin_data.cpu_block_dgemm = 0;*/
+#ifdef PASTIX_WITH_CUDA
         sopalin_data.cublas_handle = pastix_data->cublas_handle;
         sopalin_data.cublas_stat = pastix_data->cublas_stat;
+#endif
         //cublasSetMathMode(*(pastix_data->cublas_handle), CUBLAS_TENSOR_OP_MATH);
 
         /* TODO: might change the behavior: if the user wants a ratio of the norm, it could compute it himself */
