--- refinement/pastix_task_refine.c.orig	2021-09-22 17:21:31.000000000 +0200
+++ refinement/pastix_task_refine.c	2024-03-24 15:29:22.642950000 +0100
@@ -26,9 +26,11 @@
 #include <blend/solver.h>
 #include <spm.h>
 #include <bcsc/bcsc.h>
+#if defined(PASTIX_WITH_CUDA)
 #include <cuda_runtime_api.h>
 #include <cuda.h>
 #include "kernels/gpus/LightSpMV-1.0/src/cLightSpMV.h"
+#endif
 
 #include <parsec.h>
 #include <parsec/data.h>
@@ -38,7 +40,9 @@
 #endif
 #include "parsec/utils/zone_malloc.h"
 
+#if defined(PASTIX_WITH_CUDA)
 extern gpu_device_t* gpu_device;
+#endif
 extern char* gpu_base;
 
 /**
