--- refinement/z_refine_gmres_gpu.c.orig	2024-03-23 21:17:16.188998000 +0100
+++ refinement/z_refine_gmres_gpu.c	2024-03-23 21:21:02.418673000 +0100
@@ -20,8 +20,10 @@
 #include "cblas.h"
 #include "bcsc.h"
 #include "z_refine_functions.h"
+#ifdef PASTIX_WITH_CUDA
 #include <cuda_runtime_api.h>
 #include <cuda.h>
+#endif
 #include "pastix/order.h"
 
 /**
@@ -406,7 +408,7 @@
 #endif
 		}
 		else{
-			cblas_ztrsv( CblasColMajor, CblasUpper, CblasNoTrans, CblasNonUnit, i+1, (cuDoubleComplex*) gmH, im1, (cuDoubleComplex*) gmG, 1 );
+			cblas_ztrsv( CblasColMajor, CblasUpper, CblasNoTrans, CblasNonUnit, i+1, (pastix_complex64_t*) gmH, im1, (pastix_complex64_t*) gmG, 1 );
 		}
 
         /**
